### PR TITLE
feat: support multi hosted-domain in GoogleIdentityProvider

### LINKED ISSUE #16787: support multi hosted-domain in `GoogleIdentityProvider`
### Description

I'm in a group that have multi-companies, each have its own email domain like  `@companyA.com`, `@companyB.com`.
I'd like to use Google authen with keycloak to provide better user experiment. However, currently keycloak only support single domain via `hd` parameter.

It would be nice if Keycloak support **multiple hosted-doamin**, seperated by `,` like some other products like GitLab via [OmniAuth](https://github.com/zquestz/omniauth-google-oauth2/blob/00e2d0a9f9c73e463020fe2d219ff83e3c3e4790/lib/omniauth/strategies/google_oauth2.rb#L247)

### PR DESCRIPTION
**Pull request by dungdm93**
I'm in a group that have multi-companies, each have its own email domain like  `@companyA.com`, `@companyB.com`.
I'd like to use Google authen with keycloak to provide better user experiment. However, currently keycloak only support single domain via `hd` parameter.

It would be nice if Keycloak support **multiple hosted-doamin**, seperated by `,` like some other products like GitLab via [OmniAuth](https://github.com/zquestz/omniauth-google-oauth2/blob/00e2d0a9f9c73e463020fe2d219ff83e3c3e4790/lib/omniauth/strategies/google_oauth2.rb#L247)

Close #16787

### CODE REVIEW

File: services/src/main/java/org/keycloak/social/google/GoogleIdentityProvider.java [hunk 1]
```diff
@@ -100,11 +101,11 @@ protected UriBuilder createAuthorizationUrl(AuthenticationRequest request) {
         if (hostedDomain != null) {
             uriBuilder.queryParam(OIDC_PARAMETER_HOSTED_DOMAINS, hostedDomain);
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/social/google/GoogleIdentityProvider.java:102]: How does Google behave when setting a comma-separated list of domains?

**Review by pedroigor — CHANGES_REQUESTED** [review]
Thanks, @dungdm93.

Do you mind looking at this [guide](https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md) and following the first steps there to contribute to the project?

Basically, you should create an issue and link it to this PR as per the guide above.

Regarding your changes, we should also update the tooptip message for the `hostedDomain` configuration attribute to mention that a comma-separate list is allowed when setting the hosted domains.

File: services/src/main/java/org/keycloak/social/google/GoogleIdentityProvider.java (hunk 1 above)
Comment (dungdm93) [reply services/src/main/java/org/keycloak/social/google/GoogleIdentityProvider.java:102]: Glad you asked. According to [this docs](https://developers.google.com/identity/openid-connect/openid-connect#hd-param), all `hd` param does is show in the UI that email should ends with that domain.
image
And off-course, you can enter other email domain like `example@otherdomain.com`. All validation must be done in client side (`keycloak` in this case).

So, in the case of multi-domains, google simply don't show the suggestion as picture above.

**Comment by dungdm93** [comment]
Thanks @pedroigor
Issue is created, and I also update the tooltip. Could you please take a look.

**Comment by pedroigor** [comment]
@dungdm93 Thanks, running a build now.

**Comment by jonkoops** [comment]
Please also open a PR on the [new Admin UI](https://github.com/keycloak/keycloak-ui), we're pulling the plug on the AngularJS based one soon.

**Comment by dungdm93** [comment]
@jonkoops keycloak-ui PR is created
keycloak/keycloak-ui#4315

**Comment by pedroigor** [comment]
Running after rebasing.

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
@dungdm93 Thanks.

**Pull request merged**